### PR TITLE
✨ NEW: Do not allow blank lines within $$

### PR DESCRIPTION
**Pull request by rowanc1**
* This replaces #6 

From @eric-wieser:
> This matches the behavior of LaTeX (and of the stackexchange and github markdown parsers), and prevents runaway parsing if a $$ is opened but not closed.

### CODE REVIEW

**Comment by rowanc1** [comment]
@eric-wieser, I fixed up the linting and the formatting in one test, but it looks like there is an issue with:

```
$$
a
$$ (label)
```

Which I believe should still work. Maybe this is just the label part?

**Comment by eric-wieser** [comment]
That test is from

https://github.com/executablebooks/markdown-it-dollarmath/blob/8bfc9c44abd0240b5865cdcf4bbbeb55c76878ba/tests/fixtures.spec.ts#L62

Right?

File: src/index.ts [hunk 1]
```diff
@@ -304,14 +304,17 @@ function math_block_dollar(
         start = state.bMarks[nextLine] + state.tShift[nextLine]
         end = state.eMarks[nextLine]
         if (end - start < 2) {
-          continue
+          break // blank lines are not allowed within $$
```
Comment (eric-wieser) [inline src/index.ts]: This logic is wrong, it fails on lines consisting of a single character. I think I assumed that \n was counted in the total.

Comment (eric-wieser) [reply src/index.ts]: I think the entire `if` can be safely removed, as the new if replaces it

Comment (rowanc1) [reply src/index.ts]: That does fix the tests. I will push and then get a sanity check from @chrisjsewell.

**Review by rowanc1 — COMMENTED** [review]
This moves the if statement based on empty text rather than a token length, which is a bit more robust to other cases, which are now added.

Looks all good from my perspective and passes tests!

**Comment by rowanc1** [comment]
Thanks for your contribution here @eric-wieser!

File: src/index.ts (hunk 1 above)
Comment (eric-wieser) [reply src/index.ts]: Any update on this? I'd like to port a documentation tool for the Lean programming language from using `mistletoe` to useing markdown-it, but can't do so without this fix!

Comment (rowanc1) [reply src/index.ts]: I haven't heard from @chrisjsewell in the last week. I did ping him on slack. I unfortunately don't have npm admin rights to update this at the moment.

**Comment by chrisjsewell** [comment]
Seems reasonable, but should also be pushed to https://github.com/executablebooks/mdit-py-plugins/blob/master/mdit_py_plugins/dollarmath/index.py

**Comment by eric-wieser** [comment]
Is that something you port manually, or something that these is a script to port?

**Comment by chrisjsewell** [comment]
Heya, no just manually

**Comment by eric-wieser** [comment]
@rowanc1, would you be able to take a look at https://github.com/executablebooks/mdit-py-plugins/pull/46? I was hoping to use this from python, but the two versions are still not in sync.

**Comment by rowanc1** [comment]
Looks like it just needs a deploy, right?!

Yep I can test it and get it out! 🚀

**Comment by rowanc1** [comment]
Oh, I see looks like you need the merge in the python side -- @chrisjsewell any chance that you can give this a bit of attention or guidance on how to bring it in?

**Pull request merged**